packages/ferreiro-server/setup/setupMiddlewares.js

```diff
@@ -18,11 +27,56 @@ module.exports = (app) => {
   app.use(compression())
 
   // Search engines
-  app.get('/robots.txt', (req, res) => {
+  app.get('/robots.txt', crawlersRateLimit, (req, res) => {
     res.sendFile(path.join(__dirname + '/../robots.txt'))
   })
-  app.get('/sitemap.xml', (req, res) => {
-    res.sendFile(path.join(__dirname + '/../sitemap.xml'))
+
+  app.get('/sitemap.xml', crawlersRateLimit, (req, res) => {
```

Comment:
Is it possible instead to generate this at build time and deploy it as part of the bundle? Then you don't need to continuously regenerate it. The reason to not do this if there is a way to dynamically change/add urls. If that is the case another option could be to create a webhook that generates the file as a result of an action(create post, delete post)

Reply:
I think what you have is great and will work, but may be more performant to run it less.